Add documentation for Traces beta feature (#7004)
dagger/dagger: docs/current_docs/manuals/user/cloud/741031-user-interface.mdx

---

# User Interface
:::info

> Dagger 0.11.0 adds support for visualizing Dagger Functions calls in Dagger Cloud. With this update, Dagger Cloud will default to the use the new Traces experience for both `dagger call` and `dagger run` commands. Engines running Dagger 0.11.0 or later will automatically use the new Traces experience. Engines running Dagger 0.10.0 or earlier will continue to use the legacy Runs experience.

Reviewer: ```suggestion
Dagger 0.11.0 adds support for visualizing Dagger Function calls in Dagger Cloud. With this update, Dagger Cloud will default to use the new Traces experience for both `dagger call` and `dagger run` commands. Dagger v0.11.0 or later will automatically use the new Traces experience. Older versions of Dagger will continue to use the legacy Runs experience.
```

:::

> Dagger Cloud visualizes your Dagger pipelines runs invoked from your local dev machine and in CI. 

Reviewer: ```suggestion
Dagger Cloud visualizes your Dagger pipeline runs invoked from your local dev machine and in CI. 
```

A run is an invocation of a Dagger pipeline. When you run Dagger in CI, each call aligns with one run in Dagger Cloud. If you have many CI jobs configured to run for every pull/merge request, you may see several runs for a single commit.

For instance, if 4 GitHub Actions workflows each call Dagger to run a pipeline for pull requessts, you will see four distinct runs in the Dagger Cloud UI. You may also see multiple runs if one GitHub workflow has multiple jobs defined with distinct calls to Dagger to run your pipelines. As you Daggerize your pipelines, you will likely have fewer distinct CI workflows.

> Because Dagger Cloud understands the Dagger engine's data model, it gives you visibility into the steps of your pipeline's Directed Acyclic Graph (DAG), including detailed logs for each step.

Reviewer: ```suggestion
Because Dagger Cloud understands the Dagger Engine's data model, it gives you visibility into the steps of your pipeline's Directed Acyclic Graph (DAG), including detailed logs for each step.
```

## Traces
:::info

> You will only see runs in the Traces view if you are using Dagger 0.11.0 or later. If you are using an engine version before 0.11.0, you will see runs in the Runs view.

Reviewer: ```suggestion
You will only see runs in the Traces view if you are using Dagger 0.11.0 or later. If you are using an older version of Dagger, you will see runs in the Runs view.
```

:::

:::note
Dagger **Traces** is still under development and is in Beta. We welcome your feedback on how we can improve this page. Please join the [Dagger Discord](https://discord.gg/dagger-io) to join the Dagger community and share your feedback.
:::

> The **Traces** page lists runs you've kicked off from your local dev machine or in CI.

Reviewer: ```suggestion
The **Traces** page lists runs you've triggered from your local dev machine or in CI.
```

(ADD SCREENSHOT OF TRACES PAGE HERE)

Here's a quick summary of what you'll see for each trace.

|Field | Description |
|---|---|
|Status icon | Indication of run success or failure |
|Command | The exact command used to invoke your pipeline |
|Start | Run start time |
|Duration | Run duration |

:::info
The Traces view does not yet include metadata for the user, branch, remote, or commit. We are working on adding this metadata to the Traces view.
:::

### Trace Details page
You can drill down into the details of a specific trace by clicking it. This directs you to a run-specific **Trace Details** page. 

:::info

> Dagger Traces is still under development and is in Beta. We welcome your feedback on how we can improve this page. Please join the [Dagger Discord](https://discord.gg/dagger-io) to join the Dagger community and share your feedback.

Reviewer: This seems to be repeated?

:::

(ADD SCREENSHOT OF TRACE DETAILS PAGE HERE)

The **Trace Details** page includes detailed status and duration metadata about the pipeline steps. The tree view shows Dagger pipelines and steps within those pipelines. If there are any errors in the run, Dagger Cloud automatically brings you to the first error in the list. You see detailed logs and output of each step, making it easy for you to debug your pipelines and collaborate with your teammates.

A Dagger trace shows a detailed timeline for operations completed in a pipeline run. Dagger Cloud also applies unique IDs and icons for related operations, which are essential for debugging your pipeline when something does wrong.

### All Runs page
:::info

> The All Runs view only applies to Dagger pipeline runs done with engines before version 0.11.0. This view has been replaced by the new Traces view in Dagger Cloud.

Reviewer: ```suggestion
The All Runs view only applies to pipeline runs done with Dagger versions older than v0.11.0. This view has been replaced by the new **Traces** view in Dagger Cloud.
```

Because Dagger Cloud understands the Dagger engine's data model, it gives you visibility into the steps of your pipeline's Directed Acyclic Graph (DAG), including detailed logs for each step.

## Traces
:::info
You will only see runs in the **Traces** view if you are using Dagger 0.11.0 or later. If you are using an older version of Dagger, you will see runs in the **Runs** view.
:::

:::note
**Traces** is still under development and is in Beta. We welcome your feedback on how we can improve this page. Please join the [Dagger Discord](https://discord.gg/dagger-io) to join the Dagger community and share your feedback.
:::

### All Traces page
The **Traces** page lists runs you've triggered from your local development host or in CI.

![View Traces](/img/current_docs/manuals/user/cloud/user-interface/all-traces.png)

Here's a quick summary of what you'll see for each trace.

|Field | Description |
|---|---|

> |Status icon | Indication of run success or failure |

Reviewer 2: ```suggestion
|Status icon | Indication of run processing, success or failure |
```
I don't know if `processing` is the best word. `run running` is redundant. Your call @d3rp3tt3

Author: I'll use "Indication of the runs status: in progress, succeeded, or failed"

The **Trace Details** page includes detailed status and duration metadata about the pipeline steps. The tree view shows Dagger pipelines and steps within those pipelines. If there are any errors in the run, the UI shows the errors at the top. Clicking any of the errors takes you directly to the step that failed, making it easy to debug your pipelines and collaborate with your teammates.  

![View Trace details](/img/current_docs/manuals/user/cloud/user-interface/trace-detail-error-list.png)

A Trace shows a detailed timeline for operations completed in a pipeline run. You see detailed logs and output of each step, making it easy for you to debug your pipelines and collaborate with your teammates. Dagger Cloud also applies unique IDs and icons for related operations, which are essential for debugging your pipeline when something does wrong.

![View Trace Timeline](/img/current_docs/manuals/user/cloud/user-interface/trace-detail-error-timeline.png)

## Runs
:::info
The Runs views only apply to pipeline runs done with Dagger versions older than v0.11.0. This view has been replaced by the new [**Traces**](#traces) view in Dagger Cloud.
:::

### All Runs page

> The **All Runs** page lists runs you've kicked off from your local dev machine and CI.

Reviewer: ```suggestion
The **All Runs** page lists runs you've kicked off from your local development host and in CI.
```
